src/main/java/bio/terra/service/dataset/dao/DatasetDao.java

```diff
@@ -515,4 +536,29 @@ public DatasetSummary mapRow(ResultSet rs, int rowNum) throws SQLException {
                 .createdDate(rs.getTimestamp("created_date").toInstant());
         }
     }
+
+    // -- public access to other dao objects --
+
+    /**
+     * Create a new AssetSpecification. If you try to create an asset with the same name as an existing
+     * one for the same dataset, this method throws an InvalidAssetException.
+     *
+     * @param assetSpecification the AssetSpecification being created
+     * @param datasetId          the ID of the dataset corresponding to the AssetSpecification being created
+     * @return
+     */
+    @Transactional(propagation = Propagation.REQUIRED, isolation = Isolation.SERIALIZABLE)
+    public UUID createAsset(AssetSpecification assetSpecification, UUID datasetId) {
+        return assetDao.createAsset(assetSpecification, datasetId);
```

Comment:
there's a question I want to confirm: are direct calls to annotation methods transactions?

In the docs (https://docs.spring.io/spring-framework/docs/5.1.8.RELEASE/spring-framework-reference/data-access.html#transaction-declarative-annotations) they say:

```In proxy mode (which is the default), only external method calls coming in through the proxy are intercepted. This means that self-invocation (in effect, a method within the target object calling another method of the target object) does not lead to an actual transaction at runtime even if the invoked method is marked with @Transactional. Also, the proxy must be fully initialized to provide the expected behavior, so you should not rely on this feature in your initialization code (that is, @PostConstruct).```

I *think* it's ok since since the @Transactional annotation triggers transactionality within that thread and child invocations should be covered but it's worth double checking